src/main/kotlin/com/example/probation/service/impl/UsersServiceImpl.kt

```
@@ -0,0 +1,173 @@
+            enabled = true
+            usersRepository.save(this)
+        }.also { sendLog(Actions.UNBLOCK.action, Current.CURRENT.value, it.username) }
+    }
+
+    override fun deleteUser(userId: Long) {
+        getUserById(userId).let {
+            tokenService.deleteTokenByUser(it)
+            usersRepository.deleteById(userId)
+            sendLog(Actions.DELETE.action, Current.CURRENT.value, it.username)
+            sendLog(Actions.DELETED.action, it.username, Current.CURRENT.value)
+        }
+    }
+
+    override fun checkUniqueNewName(newName: String, id: Long) =
+        getUserById(id).username.let {
+            newName == it || !checkUsernameExistence(newName)
+        }
+
+    override fun checkUniqueNewEmail(newEmail: String, id: Long) =
+        getUserById(id).email.let {
+            newEmail == it || !checkEmailExistence(newEmail)
+        }
+
+    override fun changePersonalData(user: User) {
+        user.id?.let { userId ->
+            getUserById(userId).let {
+                it.id = user.id
+                it.username = user.username
+                it.description = user.description
+                it.email = user.email
+                usersRepository.save(it)
+            }
+        }
+        sendLog(Actions.EDIT.action, Current.CURRENT.value, null)
+    }
+
+    override fun changePassword(changePasswordDto: ChangePasswordDto) {
+        val user: User = getUserById(changePasswordDto.id)
+
+        if (passwordEncoder.matches(changePasswordDto.oldPassword, user.password)) {
+            user.password = passwordEncoder.encode(changePasswordDto.newPassword)
+            usersRepository.save(user)
+        } else {
+            throw PasswordDoesNotMatchesException("{password.not.matches}")
+        }
+
+        sendLog(Actions.CHANGE_PASSWORD.action, Current.CURRENT.value, null)
+    }
+
+    private fun searchCurrentUsernameIfExist(name: String?) =
+        if (name == Current.CURRENT.value) {
```

some strange logic here, it should be reworked or even removed.
first of all, you should throw the exception immediately and not catch
second, you do not need to call details service many times, just do it once and use the username from context. there is no need to compare the username with CURRENT constant